### ISSUE TITLE
About the NMS

### ISSUE THREAD

**Issue by DrugRui** [potterhsu/easy-faster-rcnn.pytorch#9]
Thanks for your clean and useful code implementation.I would like to consult you why 'class_bboxes' on the NMS are not arranged in descending order of 'class_probs' in 'model.py' when validating or testing. The code is as follows:
`kept_indices = nms(class_bboxes, class_probs, threshold)`

**Comment by potterhsu** [comment]
1. First of all, the function [`nms` has handled boxes sorting](https://github.com/potterhsu/easy-faster-rcnn.pytorch/blob/2c38b785eb488cb2dc366cbbdbe269bc5bf00ef2/support/src/cuda/nms.cu#L70-L75)
2. So that in [`generate_detections` boxes doesn't require to sort before `nms` is called](https://github.com/potterhsu/easy-faster-rcnn.pytorch/blob/2c38b785eb488cb2dc366cbbdbe269bc5bf00ef2/model.py#L204)
3. You might be confused by similar procedure in [`generate_proposals` which has done sorting before `nms`](https://github.com/potterhsu/easy-faster-rcnn.pytorch/blob/2c38b785eb488cb2dc366cbbdbe269bc5bf00ef2/rpn/region_proposal_network.py#L146-L155), it's because that we want to take top N boxes and then pass to `nms` for the efficiency (note that #boxes between `generate_detections` and `generate_proposals` was different, typically <1k and >10k)

**Comment by DrugRui** [comment]
Thank you for your reply. It is useful to me

**Comment by DrugRui** [comment]
I have another question: why is the threshold of detection_probs for the verification phase very low, so that there are enough boxes involved in the calculation of map? As in the code in evaluator.py:
` kept_indices = (detection_probs > 0.05).nonzero().view(-1)`

**Comment by potterhsu** [comment]
Right, and this threshold is determined by reference to other popular repos, such as [maskrcnn-benchmark](https://github.com/facebookresearch/maskrcnn-benchmark/blob/master/maskrcnn_benchmark/config/defaults.py#L200)

**Comment by DrugRui** [comment]
Well, thanks a lot for your help.
